### PR TITLE
SSL/TLSize all the things! (convert http:// to https:// where appropriate)

### PR DESCRIPTION
**Pull request by reedloden**
Update links to use https:// where it is supported. There's probably a lot
more that could be fixed, but these are the core ones I found (especially
the download links in order to prevent MITM attacks). Note that there are
some fb.me links that will redirect to http:// even while accessed over
https://, but this seemed like the best way to fix those for now.

NOTE: Only non-third-party files were modified. There are references to
http:// URLs in vendored/third-party files, but seems appropriate to fix
upstream for those rather than editing the files.

Also, copy one image locally to the blog, as it was hotlinking to a site
that did not support https://.

Last, use youtube-nocookie.com instead of youtube.com for video embeds,
as the former doesn't try to set a cookie on load (privacy enhancement).

### CODE REVIEW

**Comment by reedloden** [comment]
@zpao / @jsfb -- this is take two of #3687

File: docs/_posts/2014-11-25-community-roundup-24.md [hunk 1]
```diff
@@ -76,20 +76,20 @@ We're counting down the days until [React.js Conf](http://conf.reactjs.com) at F
 
 ## React Meetups Around the World
 
-<blockquote class="twitter-tweet" lang="en"><p>React JS meetup having pretty good turn up rate today <a href="https://twitter.com/hashtag/londonreact?src=hash">#londonreact</a> <a href="http://t.co/c360dlVVAe">pic.twitter.com/c360dlVVAe</a></p>&mdash; Alexander Savin (@karismafilms) <a href="https://twitter.com/karismafilms/status/535152580377468928">November 19, 2014</a></blockquote>
+<blockquote class="twitter-tweet" lang="en"><p>React JS meetup having pretty good turn up rate today <a href="https://twitter.com/hashtag/londonreact?src=hash">#londonreact</a> <a href="https://t.co/c360dlVVAe">pic.twitter.com/c360dlVVAe</a></p>&mdash; Alexander Savin (@karismafilms) <a href="https://twitter.com/karismafilms/status/535152580377468928">November 19, 2014</a></blockquote>
```
Comment (zpao) [inline docs/_posts/2014-11-25-community-roundup-24.md]: These are actually copy+paste from Twitter embeds, so I don't think we should make this change.

Comment (reedloden) [inline docs/_posts/2014-11-25-community-roundup-24.md]: I thought Twitter had retroactively changed all those to https://, but I guess not. Will change back.

File: docs/tips/16-references-to-components.md [hunk 2]
```diff
@@ -25,4 +25,4 @@ var myComponentInstance = React.render(myComponentElement, myContainer);
 
 > Note:
 >
-> This should only ever be used at the top level. Inside components, let your `props` and `state` handle communication with child components, and only reference components via [refs](http://facebook.github.io/react/docs/more-about-refs.html).
+> This should only ever be used at the top level. Inside components, let your `props` and `state` handle communication with child components, and only reference components via [refs](https://facebook.github.io/react/docs/more-about-refs.html).
```
Comment (zpao) [inline docs/tips/16-references-to-components.md]: We should probably just make this a relative link (which is what we typically do, but can follow up later)

File: docs/docs/videos.md [hunk 3]
```diff
@@ -168,6 +168,6 @@ Walk-through by [Andres Suarez](https://github.com/zertosh) on how [SoundCloud](
 
 ### Introducing React Native (+Playlist) - React.js Conf 2015
 
-<iframe width="650" height="366" src="https://www.youtube.com/embed/KVZ-P-ZI6W4?list=PLb0IAmt7-GS1cbw4qonlQztYV1TAW0sCr" frameborder="0" allowfullscreen></iframe>
+<iframe width="650" height="366" src="https://www.youtube-nocookie.com/embed/KVZ-P-ZI6W4?list=PLb0IAmt7-GS1cbw4qonlQztYV1TAW0sCr" frameborder="0" allowfullscreen></iframe>
```
Comment (zpao) [inline docs/docs/videos.md]: I'm fine with this change but we're not all going to remember to do this for all future posts so in all honesty it's probably a futile effort. The fact that https://www.youtube-nocookie.com/ is a (blank) 404 page. The [Google results](ahttps://www.google.com/search?q=youtube-nocookie&ie=utf-8&oe=utf-8) are laughable and none of them are Google mentioning their support for it.

Comment (reedloden) [inline docs/docs/videos.md]: For future note, on a YouTube video, click on "Share" -> "Embed" -> "Show More" -> "Enable privacy-enhanced mode", you'll get this same youtube-nocookie.com link. [This YouTube support page](https://support.google.com/youtube/answer/171780?expand=PrivacyEnhancedMode#privacy) provides more information. Backstory is that it was added at the request of the White House so they could embed YouTube videos on whitehouse.gov without a cookie being immediately set by YouTube for loading the interstitial (before the video is actually loaded and played).

File: docs/tips/16-references-to-components.md (hunk 2 above)
Comment (reedloden) [inline docs/tips/16-references-to-components.md]: Went ahead and just fixed it everywhere. :)

File: docs/_posts/2014-11-25-community-roundup-24.md (hunk 1 above)
Comment (reedloden) [inline docs/_posts/2014-11-25-community-roundup-24.md]: Changed.

File: docs/_posts/2013-07-17-react-v0-4-0.md [hunk 4]
```diff
@@ -18,10 +18,10 @@ When you're ready, [go download it](/react/downloads.html)!
...
-* `prop` improvements: validation and default values. [Read our blog post for details...](http://facebook.github.io/react/blog/2013/07/11/react-v0-4-prop-validation-and-default-values.html)
-* Support for the `key` prop, which allows for finer control over reconciliation. [Read the docs for details...](http://facebook.github.io/react/docs/multiple-components.html)
-* Removed `React.autoBind`. [Read our blog post for details...](http://facebook.github.io/react/blog/2013/07/02/react-v0-4-autobind-by-default.html)
-* Improvements to forms. We've written wrappers around `<input>`, `<textarea>`, `<option>`, and `<select>` in order to standardize many inconsistencies in browser implementations. This includes support for `defaultValue`, and improved implementation of the `onChange` event, and circuit completion. [Read the docs for details...](http://facebook.github.io/react/docs/forms.html)
+* `prop` improvements: validation and default values. [Read our blog post for details...](/react/blog/2013/07/11/react-v0-4-prop-validation-and-default-values.html)
```
Comment (zpao) [inline docs/_posts/2013-07-17-react-v0-4-0.md]: You da best :)

**Comment by zpao** [comment]
Alright, let's do it. Dear translators (@marocchino and the KO team especially), I'm sorry! You may be able to get a bit fancy and do some processing on the patch to automate this but otherwise it might be a bit rough.

**Pull request merged**